### ISSUE TITLE
[FEATURE_REQUEST] importing a character from DreamGen

### ISSUE THREAD

**Issue by tamwile** [SillyTavern/SillyTavern#2676]
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

As expected, importing a character from a DreamGen url doesn't work, since it's not in the list of the popup.
But importing from a file also doesn't work.
DreamGen allows to export a json file with optional data for the images encoded in b64.
I guess the json fields are not standard compared to the other LLM backends.

### Describe the solution you'd like

I would like to be able to import a character from DreamGen, either via url or with a json file.
(I have also tried to download png images of the character but it seems they don't add exif metadata).

### Describe alternatives you've considered

I have tried to export from DreamGen with and without embedding the b64 images.

### Additional context

DreamGen is listed in the "Cloud LLM APIs" of the SillyTavern documentation here:
https://docs.sillytavern.app/usage/api-connections/
So i was expecting to find a way to import a character from this website.

### Priority

Medium (Would be very useful)

### Are you willing to test this on staging/unstable branch if this is implemented?

Maybe

**Comment by DreamGenX** [comment]
Hello!

I would say the main obstacle of converting from DreamGen's "scenario" ([spec here](https://github.com/DreamGenX/scenario-spec)) to "character card" is that scenarios natively support multi-character setups. Single chracter scenarios should be easy enough to convert accurately though -- I could probably expose a simple tool for that.

In the other direction (using "char" card with DreamGen), which I guess is what most people are after, it already works:

When using SillyTavern as the frontend and DreamGen API (or HF model) as the backend, you can use existing chracter cards from characterhub or elsewhere. You might want to adjust the prompt a bit in some cases (when the card uses the W++ format, for example), but it should work OK without that. Inside the DreamGen web app, we have a tool to improve the card prompt built directly into the [importer tool](https://dreamgen.com/app/tools/scenario-importer) -- but that again produces a "scenario" and not a "card".

**Comment by Wolfsblvt** [comment]
Yeah, from what I can see from exporting some chars myself and looking at the spec, this is a non-trivial import that'd need to be programmed.
Not impossible for sure, and would be kind of *nice* to have a way to import whole group chats into ST, but *someone* would have to code that.

I mean, I can see supporting DreamGen char/scenarios as a nice-to-have idea, so if someone else wants to code this, why not.
